src/ngraph_builder.cc

```diff
@@ -1351,6 +1351,177 @@ static Status TranslateConv2DBackpropInputOp(
+
+  ng::Shape input_shape = ng_input->get_shape();
+  std::map<std::string, int> format_to_int_map = {
+      {"NHWC", 0}, {"NCHW", 1}, {"NCHW_VECT_C", 1}};
+
+  int channel_dimension;
+  int num_spatial_dimensions = 2;  // H, W are spatial dimensions
+
+  switch (format_to_int_map[tf_data_format]) {
+    // NHWC
+    case 0:
+      channel_dimension = 3;
+      break;
+    // NCHW or NCHW_VEC_C
+    case 1:
+      channel_dimension = 1;
+      break;
+    default:
+      return errors::InvalidArgument(
+          "DepthToSpace supported data format is NCHW, NHWC, or NCHW_VECT_C");
+  }
+
+  // Error checking : depth must be divisible by square of the block_size
+  if (input_shape[channel_dimension] % (block_size * block_size) != 0) {
+    return errors::InvalidArgument(
+        "Input tensor's channel dimension ,", input_shape[channel_dimension],
+        " is not divisible by square of the block_size ", block_size);
+  }
+
+  ng::AxisVector ng_reshape_shape;
+  ng::AxisVector ng_transpose_shape;
+  ng::AxisVector ng_output_shape;
+
+  if (tf_data_format.compare("NCHW") == 0) {
+    ng::AxisVector input_transpose_shape;
+    input_transpose_shape.push_back(0);
+    input_transpose_shape.push_back(2);
+    input_transpose_shape.push_back(3);
+    input_transpose_shape.push_back(1);
+
+    ng_input = ng::builder::numpy_transpose(ng_input, input_transpose_shape);
+  } else if (tf_data_format.compare("NCHW_VECT_C") == 0) {
+    ng::AxisVector input_transpose_shape;
+    input_transpose_shape.push_back(0);
+    input_transpose_shape.push_back(2);
+    input_transpose_shape.push_back(3);
+    input_transpose_shape.push_back(1);
+    input_transpose_shape.push_back(4);
+
+    ng_input = ng::builder::numpy_transpose(ng_input, input_transpose_shape);
+  }
+
+  switch (format_to_int_map[tf_data_format]) {
+    // NHWC
+    case 0: {
+      ng_reshape_shape.push_back(input_shape[0]);  // N dimension
+      for (int i = 0; i < num_spatial_dimensions; i++) {
+        ng_reshape_shape.push_back(input_shape[i + 1]);
+      }
+      int64 num_blocks = 1;
+      for (int i = 0; i < num_spatial_dimensions; i++) {
+        ng_reshape_shape.push_back(block_size);
+        num_blocks *= block_size;
+      }
+      ng_reshape_shape.push_back(input_shape[channel_dimension] / num_blocks);
+
+      ng_transpose_shape.push_back(0);
+      for (int i = 0; i < num_spatial_dimensions; i++) {
+        ng_transpose_shape.push_back(i + 1);
+        ng_transpose_shape.push_back(i + 1 + num_spatial_dimensions);
+      }
+      ng_transpose_shape.push_back(channel_dimension + num_spatial_dimensions);
+
+      ng_output_shape.push_back(input_shape[0]);
+      for (int i = 0; i < num_spatial_dimensions; i++) {
+        ng_output_shape.push_back(input_shape[i + 1] * block_size);
+      }
+      ng_output_shape.push_back(input_shape[channel_dimension] / num_blocks);
+      break;
+    }  // end of case NHWC
+    // NCHW
+    case 1: {
+      cout << "shouldn't enter" << endl;
```

Comment:
Please remove the cout

Reply:
Or a more descriptive graph-log why one shouldn't enter